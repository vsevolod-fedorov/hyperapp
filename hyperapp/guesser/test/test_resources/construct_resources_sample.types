
sample_list_d = record
sample_list_open_d = record
sample_list_parent_d = record

sample = record:
  id: int

sample_item = record:
  key: int
  name: string
  description: string


sample_global_command_d = record
sample_object_command_d = record
