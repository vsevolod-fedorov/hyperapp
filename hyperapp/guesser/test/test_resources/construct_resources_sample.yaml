import:
  types:
    construct_resources_sample:
      - sample
      - sample_item
