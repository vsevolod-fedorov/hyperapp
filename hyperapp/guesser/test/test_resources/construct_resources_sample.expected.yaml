import:
- common.mark:mark.service
- legacy_service:mosaic
- transport.identity:peer_registry.service
- legacy_service:rpc_call_factory
- legacy_service:web
- legacy_type.attribute:attribute
- legacy_type.builtin:int
- legacy_type.command:object_commands_d
- legacy_type.construct_resources_sample:sample
- legacy_type.construct_resources_sample:sample_global_command_d
- legacy_type.construct_resources_sample:sample_item
- legacy_type.construct_resources_sample:sample_list_d
- legacy_type.construct_resources_sample:sample_list_open_d
- legacy_type.construct_resources_sample:sample_list_parent_d
- legacy_type.construct_resources_sample:sample_object_command_d
- legacy_type.global_command:global_command_association
- legacy_type.impl:global_command_impl
- legacy_type.impl:impl_association
- legacy_type.impl:list_spec
- legacy_type.impl:method_command_impl
- legacy_type.impl:object_command_impl
- legacy_type.builtin:python_object_association
- legacy_type.lcs:lcs_set_association
- legacy_type.builtin:python_module
associations:
- _type: legacy_type.global_command:global_command_association
  command: sample_global_command.command
- _type: legacy_type.impl:impl_association
  piece_t: legacy_type.construct_resources_sample:sample
  ctr_fn: SampleList
  spec: SampleList.spec
- _type: legacy_type.builtin:python_object_association
  t: legacy_type.construct_resources_sample:sample
  function: SampleList
- _type: legacy_type.lcs:lcs_set_association
  dir:
  - object_commands_d
  value: sample_object_command.command
- _type: legacy_type.lcs:lcs_set_association
  dir:
  - sample_list_d
  - object_commands_d
  value: SampleList.open.command
- _type: legacy_type.lcs:lcs_set_association
  dir:
  - sample_list_d
  - object_commands_d
  value: SampleList.parent.command
definitions:
  SampleList:
    _type: legacy_type.attribute:attribute
    object: construct_resources_sample.module
    attr_name: SampleList
  SampleList.open.command:
    _type: legacy_type.impl:method_command_impl
    method: open
    params:
    - current_key
    dir: sample_list_open_d
  SampleList.parent.command:
    _type: legacy_type.impl:method_command_impl
    method: parent
    params: []
    dir: sample_list_parent_d
  SampleList.spec:
    _type: legacy_type.impl:list_spec
    key_attribute: key
    key_t: legacy_type.builtin:int
    dir: sample_list_d
  construct_resources_sample.module:
    _type: legacy_type.builtin:python_module
    module_name: construct_resources_sample
    file_name: construct_resources_sample.dyn.py
    import_list:
      htypes.construct_resources_sample.sample: legacy_type.construct_resources_sample:sample
      htypes.construct_resources_sample.sample_item: legacy_type.construct_resources_sample:sample_item
      services.mark: common.mark:mark.service
      services.mosaic: legacy_service:mosaic
      services.peer_registry: transport.identity:peer_registry.service
      services.rpc_call_factory: legacy_service:rpc_call_factory
      services.web: legacy_service:web
  object_commands_d:
    _type: legacy_type.command:object_commands_d
  sample_global_command:
    _type: legacy_type.attribute:attribute
    object: construct_resources_sample.module
    attr_name: sample_global_command
  sample_global_command.command:
    _type: legacy_type.impl:global_command_impl
    function: sample_global_command
    dir: sample_global_command_d
  sample_global_command_d:
    _type: legacy_type.construct_resources_sample:sample_global_command_d
  sample_list_d:
    _type: legacy_type.construct_resources_sample:sample_list_d
  sample_list_open_d:
    _type: legacy_type.construct_resources_sample:sample_list_open_d
  sample_list_parent_d:
    _type: legacy_type.construct_resources_sample:sample_list_parent_d
  sample_object_command:
    _type: legacy_type.attribute:attribute
    object: construct_resources_sample.module
    attr_name: sample_object_command
  sample_object_command.command:
    _type: legacy_type.impl:object_command_impl
    function: sample_object_command
    params:
    - piece
    - view_state
    dir: sample_object_command_d
  sample_object_command_d:
    _type: legacy_type.construct_resources_sample:sample_object_command_d
